src/pages/patterns/tables/basic-formatting.mdx

```diff
@@ -0,0 +1,441 @@
+
+- Users should only be able to take action on what they can see
+
+![Pagination Selecting All](figma:patterns-tables-basic-formatting-pagination-selecting-all.png)
+
+#### Infinite scrolling
+
+In cases where the database supports infinite scrolling, there is no need to say how many pages of data there are.
+
+If you have an infinitely scrolling table, you should also offer [sorting](/patterns/tables/basic-formatting#sorting) to
+help users navigate the data.
+
+##### Select all for infinite scrolling
+
+In infinitely scrolling tables, you can only manually select items.
+
+##### Vertical scrolling
+
+For tables with more rows than can be displayed in the viewport, the header row is fixed at the top of the viewport.
+Keeping the column headers within view while scrolling helps users maintain context.
+
+import TableVerticalScrolling from '../examples/tables/TableVerticalScrolling.tsx';
+import TableVerticalScrollingCode from '!!raw-loader!../examples/tables/TableVerticalScrolling.tsx';
+
+<CodeExample code={TableVerticalScrollingCode}>
+  <TableVerticalScrolling />
+</CodeExample>
+
+##### Virtualization
+
+Virtualization means what you see in the browser is what is in the DOM. When something scrolls out of view, it gets
+removed from the page DOM. When something scrolls into view, it is added to the DOM.
+
+- Use [react-window](https://github.com/bvaughn/react-window) to efficiently render large data sets
+
+import TableVirtualized from '../examples/tables/TableVirtualized.tsx';
+import TableVirtualizedCode from '!!raw-loader!../examples/tables/TableVirtualized.tsx';
+
+<CodeExample code={TableVirtualizedCode}>
+  <TableVirtualized />
+</CodeExample>
+
+#### Sorting
+
+Sorting is implemented by selecting arrows in table headers.
+
+- Clicking the header text will sort by ascending order (A→Z, 1→9, Date and Time, Priority (High→Low))
+- Clicking again will reverse the order
+- Clicking a third time will remove sorting
+
+If any ordering applied to a table including default ordering, the sorting icon should be present. This indicates how
+the table is being ordered and also creates a better affordance for the control.
+
+import TableSort from '../examples/tables/TableSort.tsx';
+import TableSortCode from '!!raw-loader!../examples/tables/TableSort.tsx';
+
+<CodeExample code={TableSortCode}>
+  <TableSort />
+</CodeExample>
+
+#### Draggable
+
+T​​o have row reordering, you can apply drag-and-drop functionality with [react-beautiful-dnd](https://github.com/atlassian/react-beautiful-dnd).
+For accessibility support, please [read the react-beautiful-dnd guidelines](https://github.com/atlassian/react-beautiful-dnd/blob/master/docs/about/accessibility.md).
+
+import TableDraggable from '../examples/tables/TableDraggable.tsx';
+import TableDraggableCode from '!!raw-loader!../examples/tables/TableDraggable.tsx';
+
+<CodeExample code={TableDraggableCode}>
+  <TableDraggable />
+</CodeExample>
+
```

Comment:
Should we also have a recommendation on column reordering as I have seen that topic come up quite frequently?

Reply:
@mtomcal Until that interaction pattern is been appropriately reviewed by Garden and the Accessibility team and documented, we can start with a recommendation.